sw/device/silicon_creator/lib/shutdown.h

```diff
@@ -30,6 +30,26 @@ extern "C" {
     }                            \
   } while (false)
 
+/**
+ * Evaluate an expression and call `shutdown_finalize` if the result is false.
```

Comment:
I would add "Evaluate a manifestly true expression and call ...". Probably emphasize that this is intended to be used in places where the compiler could otherwise prove it is statically unreachable.